Fixing connection issue to localhost:3000

GoodOlChuck opened the pull request:
See issue in issue tracker.

@kevincox @ZachDFK @matthewcrooke1

kevincox committed:
Create env var to set default bind address.

kevincox commented:
Why is so much of that file duplicated?

GoodOlChuck commented:
whoops.

kevincox commented on config/boot.rb:
```diff
@@ -1,3 +1,14 @@
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
-require 'bundler/setup' # Set up gems listed in the Gemfile.
+require 'bundler/setup' if File.exist?(ENV['BUNDLE_GEMFILE'])
```
This isn't necessary. Rails always uses bundler now days.

kevincox commented on config/boot.rb:
```diff
@@ -1,3 +1,14 @@
+  class Server
+    alias :default_options_alias :default_options
+    def default_options
+      default_options_alias.merge!(:Host => '0.0.0.0')
```
Can you create and env variable instead? Something like:

``` rb
default_options_alias.merge!(Host: ENV.fetch('BIND_ADDR', 'localhost'))
```

Then set `BIND_ADDR` in vagrant somewhere. I don't want to default to `0.0.0.0` because of security reasons.

GoodOlChuck commented:
Ok. I have made the change to `config/boot.rb`, should I make a script to set the environment variables? or should that be something we create for ourselves each time?

kevincox commented:
I would set them in `/etc/profile` or `~/.profile`. The best solution is probably to create `/etc/profile.d/vagrant.sh` (or something like that) where you export the variables.

GoodOlChuck commented:
I had it working and now I am getting `<class:Server>': undefined method 'default_options' for class 'Rails::Server' (NameError)` whenever I try and run rails server. 

Do you know why I am getting this error? whenever I comment the lines i've added, it works. 

Note: fixed the typo `default_options_alias.merge!(:Host ENV.fetch('BIND_ADDR', 'localhost'))` to `default_options_alias.merge!(Host: ENV.fetch('BIND_ADDR', 'localhost'))`

kevincox commented:
Is the guide you have for the same version of rails?

GoodOlChuck commented:
The guide was for rails 4.2

kevincox commented on Vagrantfile:
```diff
@@ -43,6 +43,13 @@ Vagrant.configure(2) do |config|
-	
+
+	$env_var_script = <<SCRIPT
+	echo # vagrant profile script > /etc/profile.d/vagrant.sh
```
This won't do what you want. I would recommend putting the following in `bin/setup-vagrant.sh`

``` sh
cat <<EOS > /etc/profile.d/vagrant.sh
# script here
EOS
```

kevincox commented:
Do you want me to take this over?

GoodOlChuck commented:
I mean sure if you want to - I am not very familiar w this stuff as you can tell.. tbh I have just been grabbing solutions that I have found online, which probably is not the best approach.

kevincox commented:
I think it's working, I'm just rebuilding my box to make sure.

kevincox commented:
@GoodOlChuck Review please. It's all working!

```
vagrant@kevinvagrantblog:/vagrant$ bin/rails server
=> Booting WEBrick
=> Rails 4.2.5.1 application starting in development on http://0.0.0.0:3000
=> Run `rails server -h` for more startup options
=> Ctrl-C to shutdown server
[2016-03-09 21:00:42] INFO  WEBrick 1.3.1
[2016-03-09 21:00:42] INFO  ruby 2.3.0 (2015-12-25) [x86_64-linux]
[2016-03-09 21:00:42] INFO  WEBrick::HTTPServer#start: pid=24927 port=3000
```

GoodOlChuck commented:
We are in business. Thanks for solving this. :+1:
